webapp/src/components/expanded_view/component.tsx

```diff
@@ -336,10 +336,10 @@ export default class ExpandedView extends React.PureComponent<Props, State> {
             this.screenPlayer.current.srcObject = this.state.screenStream;
         }
 
-        const callsClient = this.getCallsClient();
-        if (!this.state.screenStream && callsClient?.getLocalScreenStream()) {
+        const localScreenStream = this.getCallsClient()?.getLocalScreenStream();
+        if (localScreenStream && (!this.state.screenStream || this.state.screenStream.getVideoTracks()[0].id !== localScreenStream.getVideoTracks()[0].id)) {
```

Comment:
Same as the first comment, not sure if it changes the semantics, but can this be simplified to:
```suggestion
        if (localScreenStream && (this.state.screenStream?.getVideoTracks()[0].id !== localScreenStream.getVideoTracks()[0].id)) {
```